utils/atomic-storybook/.storybook/main.js

```diff
@@ -1,20 +1,31 @@
 const path = require('path');
 
+const atomicDirectory = path.join(
+  require.resolve('@coveo/atomic/package.json'),
+  '..'
+);
+
+/** @type {import('@storybook/core-common').StorybookConfig} */
 module.exports = {
   core: {
     builder: 'webpack5',
   },
-  stories: ['../src/**/*.stories.mdx', '../src/**/*.stories.tsx'],
+  stories: [
+    path.join(atomicDirectory, 'src/**/*.stories.mdx'),
+    path.join(atomicDirectory, 'src/**/*.stories.tsx'),
+  ],
   addons: [
     '@storybook/addon-docs',
     '@storybook/addon-controls',
     '@storybook/addon-viewport',
     '@storybook/addon-a11y',
     './preset.js',
   ],
+  /** @type {import('webpack')['config']['getNormalizedWebpackOptions']} */
   webpack: (config) => {
     return {
       ...config,
+      resolve: {...config.resolve, modules: module.paths},
```

Comment:
`module.paths` here contains the `node_modules` directory of `@coveo/atomic-storybook`. Without it, stories would resolve their dependencies relative to `@coveo/atomic` instead.